### PR TITLE
Dynamic update of GS

### PR DESCRIPTION
**Pull request by CedricGuillemet**
PG : https://playground.babylonjs.com/#GSUTLR#8
documentation/Validation test will be done next week

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15577/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15577/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15577/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15577/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15577/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15577/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15577/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15577/merge#BCU1XR#0

**Review by RaananW — APPROVED** [review]

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15577/merge/testResults/webgpuplaywright/index.html

**Comment by CedricGuillemet** [comment]
Doing more tests ...

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts [hunk 1]
```diff
@@ -505,6 +515,9 @@ export class GaussianSplattingMesh extends Mesh {
             return new RawTexture(data, width, height, format, this._scene, false, false, Constants.TEXTURE_BILINEAR_SAMPLINGMODE, Constants.TEXTURETYPE_FLOAT);
         };
 
+        const updateTextureFromData = (texture: BaseTexture, data: Float32Array, width: number, height: number) => {
+            (this.getEngine() as ThinEngine).updateTextureData(texture.getInternalTexture()!, data, 0, 0, width, height, 0, 0, false);
```
Comment (ryantrem) [inline packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:519]: I guess `updateTextureData` should be in `AbstractEngine`? Looks like `WebGPUEngine` implements it as well?

**Review by Popov72 — APPROVED** [review]

**Pull request merged**